### PR TITLE
Disc 263/component structure in repo

### PR DESCRIPTION
**Pull request by jesperlauridsen**
First stab at a nicer structure. Feel free to go nuts here!

### CODE REVIEW

File: src/components/search/wc_facet-checkbox.ts [hunk 1]
```diff
@@ -1,27 +1,40 @@
...
+					);
+			  })
+			: null;
 
 		const observer = new IntersectionObserver((entries) => {
 			entries.forEach((entry) => {
 				if (entry.isIntersecting) {
-					this.style.transitionDelay = this.slide / 50 + 's';
-					this.style.opacity = 1;
+					this.style.transitionDelay = this.slide || 0 / 50 + 's';
```
Comment (jorntx) [inline src/components/search/wc_facet-checkbox.ts:36]: Don't quite understand what is going on here. If `this.slide` is undefined the `transitionDelay` will be `0/50 + 's'` - isn't `0 / 50 + 's'` the same as just setting it to '0s'?

**Review by jorntx — COMMENTED** [review]
I like the overall direction that this is heading - good refactor of the struture. As we talked about yesterday:

* I like the 'wc' prefix for web components and I vote for we stick to 'wc-' instead of 'wc_' so we let the prefix alone be the determining part and limit the different naming styles we have (we also kebab case for typescript classes)
* Lets keep the pascal case naming for the vue components for now.  
*

File: src/components/search/wc_facet-checkbox.ts (hunk 1 above)
Comment (jesperlauridsen) [reply src/components/search/wc_facet-checkbox.ts]: Should be fixed now!

**Comment by jorntx** [comment]
This branch now needs a merge update:)

**Review by jorntx — APPROVED** [review]
Looks good to go - give me a heads up after you merge main in and I will merge this.

**Pull request merged**